source/usr/lib/mcvirt/client/rpc.py

```diff
@@ -0,0 +1,124 @@
+"""Provide class for connecting to RPC daemon"""
+
+# Copyright (c) 2016 - I.T. Dev Ltd
+#
+# This file is part of MCVirt.
+#
+# MCVirt is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MCVirt is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MCVirt.  If not, see <http://www.gnu.org/licenses/>
+
+import Pyro4
+
+import mcvirt.exceptions # Import necessary for Pyro # noqa
+from mcvirt.exceptions import AuthenticationError
+from mcvirt.utils import get_hostname
+from mcvirt.rpc.ssl_socket import SSLSocket
+from mcvirt.rpc.constants import Annotations
+
+
+class Connection(object):
+    """Connection class, providing connections to the Pyro MCVirt daemon"""
+
+    NS_PORT = 9090
+    SESSION_OBJECT = 'session'
+
+    def __init__(self, username=None, password=None, session_id=None,
+                 host=None):
+        """Store member variables for connecting"""
+        # If the host is not provided, default to the local host
+        self.__host = host if host is not None else get_hostname()
+
+        Pyro4.config.USE_MSG_WAITALL = False
+        Pyro4.config.CREATE_SOCKET_METHOD = SSLSocket.create_ssl_socket
+        Pyro4.config.CREATE_BROADCAST_SOCKET_METHOD = SSLSocket.create_broadcast_ssl_socket
+        self.__username = username
+        self.__ignore_drbd = False
+        self.__ignore_cluster = False
+        if 'proxy_user' in dir(Pyro4.current_context):
+            self.__proxy_username = Pyro4.current_context.proxy_user
+        else:
+            self.__proxy_username = None
+
+        # Store the passed session_id so that it may abe used for the initial connection
+        self.__session_id = session_id
+
+        # Perform an initial connection to obtain/verify the session ID
+        self.__session_id = self.__get_session(password=password)
+
+    def __get_session(self, password):
+        """Obtain a session ID"""
+        try:
+            # Attempt to obtain a connection and obtain a session ID
+            session_object = self.get_connection(object_name=self.SESSION_OBJECT,
+                                                 password=password)
+            session_id = session_object._pyroHandshake['SEID']
```

Comment:
Fixed in https://github.com/ITDevLtd/MCVirt/commit/7939a2953d6654fc3db1136901e8b42b3c18a8f0